lms/static/scripts/frontend_apps/services/grading.js

```diff
@@ -0,0 +1,66 @@
+import { apiCall } from '../utils/api';
+
+/**
+ * @typedef Student
+ * @prop {string} LISResultSourcedId - Unique outcome identifier
+ * @prop {string} LISOutcomeServiceUrl - API URL for posting outcome results
+ */
+
+/**
+ * The fetched grade success result.
+ *
+ * @typedef FetchGradeResult
+ * @prop {number} currentScore - The fetched grade
+ */
+
+/**
+ * Service for fetching and reporting student grades for an assignment.
```

Comment:
The verb `reporting` here is a little bit distracting, do we really mean `posting`?

Reply:
"Posting" is correct in the sense of the HTTP verb, although we use the verb "submit" in the method names (submitGrade) and component (`SubmitGradeForm`), so I think that would be better for consistency.